modules/apps/calendar/calendar-service/src/main/java/com/liferay/calendar/internal/util/CalendarICalDataHandler.java

```diff
@@ -628,7 +630,20 @@ private net.fortuna.ical4j.model.Calendar _toICalCalendar(
 		List<CalendarComponent> calendarComponents =
 			iCalCalendar.getComponents();
 
+		Set<String> timeZoneIds = new HashSet<>();
+
 		for (CalendarBooking calendarBooking : calendarBookings) {
+			TimeZone timeZone = calendarBooking.getTimeZone();
+
+			String timeZoneId = timeZone.getID();
+
+			if (timeZoneIds.add(timeZoneId)) {
```

Comment:
@mbowerman, maybe this will be asked:

````java
if (timeZoneIds.add(timeZone.getID())) {
````

instead of:

````java
String timeZoneId = timeZone.getID();

if (timeZoneIds.add(timeZoneId)) {
````

anyway, your fix looks good to me and I've checked the failures and they seem not related,

I'll ask for BPM's QA team double check and after that we can forward.

Thank you!

Reply:
Oh right, originally I was reusing the timeZoneId later, but now I'm not doing that anymore, so we can simplify it how you described.